jobAPI_swagger.yaml

```diff
@@ -0,0 +1,378 @@
+      requestBody:
+        description: Update an existing job in the system
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Job'
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Job'          
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Job not found
+        '422':
+          description: Validation exception
+  /jobs:
+    get:
+      tags:
+        - job
+      summary: Get jobs with filters
+      description: Returns a list of jobs based on filters
+      operationId: getJobs
+      parameters:
+        - name: categoryId
+          in: query
+          description: Filter by category ID
+          schema:
+            type: integer
+            format: int64
+        - name: levelId
+          in: query
+          description: Filter by level ID
+          schema:
+            type: integer
+            format: int64
+        - name: companyName
+          in: query
+          description: Filter by company name
+          schema:
+            type: string
+        - name: popular
+          in: query
+          description: Filter by popular jobs
+          schema:
+            type: boolean
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Job'
+        '404':
+          description: No jobs found for the given filters
+  /job/submit:
+    post:
+      tags:
+        - job
+      summary: Submit a job for admin approval
+      description: Allows users to submit a job listing for admin approval
+      operationId: submitJobForApproval
+      requestBody:
+        description: Job details to be submitted for approval
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Job'
+      responses:
+        '202':
+          description: Job submitted for approval
+        '400':
+          description: Invalid input
+        '422':
+          description: Validation exception
+  /job/{jobId}/publish:
+    post:
+      tags:
+        - admin
+      summary: Publish job on all communication channels
+      description: Send job to all communication channels (e.g., WhatsApp groups)
+      operationId: publishJob
+      parameters:
+        - name: jobId
+          in: path
+          description: ID of job to publish
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Job published successfully
+        '404':
+          description: Job not found
+  /job/{jobId}/approve:
```

Comment:
drop this endpoint
Admin will just send a POST /jobs/id with the approval state